### DISCUSSION TITLE
Linux compatibility

### DISCUSSION THREAD

**Discussion by stevehayles** [StephenHidem/AntPlus#29] [General]
Given that Garmin have released a beta version of the ANT SDK for Linux it looks possible to extend this library to work on Linux platforms.  The C++ native wrapper and managed library are missing from the Linux SDK but much of the source code looks very similar across both hardware platforms.  I was wondering if anyone had looked at this,  I'm very conversant in Dotnet and use it a lot on Linux but the C++ interfacing looks a little harder ?

**Comment by StephenHidem** [comment]
I don't have a lot of experience with working on Linux projects, but I did download the Linux beta package from thisisant. What I think you are looking for is a bridge to the .NET world. P/Invoke offers that. You could create a csharp class decorated with the DllImport attributes to access functions in the ANT Linux libraries from your .NET app. For example -

`        [DllImport(ANT_DYLIB, CallingConvention=CallingConvention.Cdecl)]
        private static extern int ANT_Init(byte ucUSBDeviceNum, UInt32 usBaudrate, ref IntPtr returnSerialPtr, ref IntPtr returnFramerPtr, byte ucPortType, byte ucHCIFrameTpye);

See this link for guidance - https://developers.redhat.com/blog/2019/03/25/using-net-pinvoke-for-linux-system-functions#
